tests/suites/os/tests/purge-data/index.js

```diff
@@ -87,90 +69,86 @@ module.exports = {
-				await this.utils.waitUntil(async () => {
-					return (
-						(await this.context
-							.get()
-							.worker.executeCommandInHostOS(
-								`systemctl is-active balena-supervisor.service`,
-								this.link,
-							)) === 'active'
-					);
-				}, false, 2 * 60 * 4, 250);
+				await test.resolves(
+					this.systemd.waitForServiceState(
+						'balena-supervisor.service',
+						'active',
+						this.link
+					),
+					'Should wait for balena.service to be active'
+				);
+
+				await test.resolves(
+					this.systemd.waitForServiceState(
+						'balena-supervisor.service',
+						'active',
+						this.link
+					),
+					'Should wait for balena-supervisor.service to be active'
+				);
 
 				test.is(
-					await this.context
-						.get()
-						.worker.executeCommandInHostOS(
+					await this.worker.executeCommandInHostOS(
 							`/usr/lib/balena/balena-healthcheck >/dev/null 2>&1 ; echo $?`,
 							this.link,
 						),
 					'0',
 					'Engine healthcheck should pass.',
 				);
 
-				test.comment(`Writing test file to data partition...`);
-				await this.context
-					.get()
-					.worker.executeCommandInHostOS(
-						`touch ${testFile}`,
+				test.is(
+					await this.worker.executeCommandInHostOS(
+						`touch ${testFile} ; echo $?`,
 						this.link,
-					);
+					),
+					'0',
+					`Should write test file ${testFile}`
+				);
 
-				test.comment(`Clearing reset flag from data partition...`);
-				await this.context
-					.get()
-					.worker.executeCommandInHostOS(
-						`rm ${resetFile}`,
+				test.is(
+					await this.worker.executeCommandInHostOS(
+						`rm ${resetFile} ; echo $?`,
 						this.link,
-					);
+					),
+					'0',
+					`Should clear reset file ${resetFile}`
+				);
 
 				// reboot
 				await this.worker.rebootDut(this.link);
 
-				test.comment(`Waiting for engine service to be active...`);
-				await this.utils.waitUntil(async () => {
-					return (
-						(await this.context
-							.get()
-							.worker.executeCommandInHostOS(
-								`systemctl is-active balena.service`,
-								this.link,
-							)) === 'active'
-					);
-				}, false, 2 * 60 * 4, 250);
-
-				test.comment(`Waiting for supervisor service to be active...`);
-				await this.utils.waitUntil(async () => {
-					return (
-						(await this.context
-							.get()
-							.worker.executeCommandInHostOS(
-								`systemctl is-active balena-supervisor.service`,
-								this.link,
-							)) === 'active'
-					);
-				}, false, 2 * 60 * 4, 250);
-
-				await waitUntilSupervisorActive(test, this.context.get());
+				await test.resolves(
+					this.systemd.waitForServiceState(
+						'balena-supervisor.service',
+						'active',
+						this.link
+					),
+					'Should wait for balena.service to be active'
+				);
+
+				await test.resolves(
+					this.systemd.waitForServiceState(
+						'balena-supervisor.service',
+						'active',
+						this.link
+					),
+					'Should wait for balena-supervisor.service to be active'
+				);
```

Comment:
```suggestion
				await test.resolves(
					this.systemd.waitForServiceState(
						'balena.service',
						'active',
						this.link
					),
					'Should wait for balena.service to be active'
				);

				await test.resolves(
					this.systemd.waitForServiceState(
						'balena-supervisor.service',
						'active',
						this.link
					),
					'Should wait for balena-supervisor.service to be active'
				);
```

Same thing twice, can we create a local helper to check these things? 
Also fixed the earlier issue of SV service being checked twice